contrib/ssh-copy-id

```diff
@@ -295,9 +299,22 @@ case "$REMOTE_VERSION" in
     populate_new_ids 0
     # in ssh below - to defend against quirky remote shells: use 'exec sh -c' to get POSIX;
     #     'cd' to be at $HOME; add a newline if it's missing; and all on one line, because tcsh.
-    [ "$DRY_RUN" ] || printf '%s\n' "$NEW_IDS" | \
+    if ! [ "$DRY_RUN" ] && ! [ "$USE_SFTP" ]; then
+      printf '%s\n' "$NEW_IDS" | \
       ssh "$@" "exec sh -c 'cd ; umask 077 ; mkdir -p .ssh && { [ -z "'`tail -1c .ssh/authorized_keys 2>/dev/null`'" ] || echo >> .ssh/authorized_keys ; } && cat >> .ssh/authorized_keys || exit 1 ; if type restorecon >/dev/null 2>&1 ; then restorecon -F .ssh .ssh/authorized_keys ; fi'" \
       || exit 1
+    fi
+    if ! [ "$DRY_RUN" ] && [ "$USE_SFTP" ] ; then
+      ssh -f -N -M -S /tmp/ssh-copy-id-shared-connection "$@"
+      # todo: create .ssh/authorized_keys if not exist
+      # todo: grep the output of the sftp-commands and exit 1 on failures
+      echo "get .ssh/authorized_keys /tmp/ssh-copy-id_authorized_keys" | sftp -o "ControlPath=/tmp/ssh-copy-id-shared-connection" "notnecessary"
+      [ -z "$(tail -1c /tmp/ssh-copy-id_authorized_keys 2>/dev/null)" ] || echo >> /tmp/ssh-copy-id_authorized_keys
```

Comment:
This line deserves a comment imo
```suggestion
      # Ensure a newline separates existing IDs from new additions
      [ -z "$(tail -1c /tmp/ssh-copy-id_authorized_keys 2>/dev/null)" ] || echo >> /tmp/ssh-copy-id_authorized_keys
```

Reply:
this cemment is already done a few lines above. when the todos are done, the comments will be at a place where they belong

Reply:
Okay.